agent/agents/supervisor/supervisor.go

```diff
@@ -359,6 +359,12 @@ func (s *Supervisor) setBuiltinAgents(builtinAgents map[string]*agentpb.SetState
 		<-agent.done
 
 		delete(s.builtinAgents, agentID)
+
+		agentTmp := filepath.Join(s.cfg.Get().Paths.TempDir, strings.ToLower(agent.requestedState.Type.String()), agentID)
```

Comment:
Are we sure that Paths cannot be nil?

Reply:
Pretty sure. If the agent is about to be stopped, which means it is running, its `cfg.Paths` must exist.